src/org/mozartoz/truffle/nodes/builtins/OSBuiltins.java

```diff
@@ -206,9 +207,15 @@ protected OzNode castMode(OzNode value) {
 
 		@Specialization
 		Object fopen(String fileName, String mode) {
-			assert mode == "rb";
 			try {
-				return new FileInputStream(new File(fileName));
+				switch (mode) {
+				case "rb":
+					return new FileInputStream(new File(fileName));
+				case "wb":
+					return new FileOutputStream(new File(fileName));
+				default:
+					throw new OzException(this, OzException.newSystemError(ERROR_FACTORY.newInstance("os", "os", "fopen", 2, "Opening mode not implemented")));
```

Comment:
A bit of a long line :wink: (I would extract the error record in a local variable)